hermes/probe/create/create.go

```diff
@@ -155,27 +157,38 @@ func CreateFile(ctx context.Context, target *probe.Target, fileID int32, fileSiz
 			status = metrics.ProbeFailed
 		}
 		target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
-		return fmt.Errorf("CreateFile(id: %d).%q: could not create file %q: %v", fileID, status, fileName, err)
+		return fmt.Errorf("CreateFile(id: %d).%q: could not create file %q: %w", fileID, status, fileName, err)
 	}
 	if err := wc.Close(); err != nil {
 		status = metrics.WriterCloseFailed
 		target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
-		return fmt.Errorf("Writer.Close: %v with status %q", err, status)
+		return fmt.Errorf("Writer.Close: %w with status %q", err, status)
 	}
 	status = metrics.Success
 	target.LatencyMetrics.APICallLatency[metrics.APICreateFile][status].Metric(hermesAPILatencySeconds).AddFloat64(time.Now().Sub(start).Seconds())
 
 	// Verify that the file that has just been created is in fact present in the target system
-	query := &storage.Query{Prefix: fileName[:FileNamePrefixSize]}
+	query := &storage.Query{Prefix: fmt.Sprintf(FileNameFormat, fileID, "")}
 	start = time.Now()
 	objIter := client.Bucket(bucketName).Objects(ctx, query)
 	finish := time.Now()
```

Comment:
Move the finish to where it actually finishes.

Reply:
This is the APILatency that's why the finish is here

Reply:
The API operation in this case is "looking up all filenames matching a query".
That includes iterating over all items, as each iteration might incur a round-trip to the service.
This means the API operation is only "done" when we the iteration is done.

Reply:
Okay thank you for explaining this changed!